### PR TITLE
chore: add license header to source files

### PR DESCRIPTION
**Pull request by trentm**
This replaces #1748. Credit to @vigneshshanmugam there for the idea to use an eslint rule for this. I'm using the existing https://github.com/nikku/eslint-plugin-license-header plugin to do this.

Closes: #1718

### CODE REVIEW

**Comment by trentm** [comment]
The current "Test" check failure is out of date. It is reporting a result from build 6, even though we currently have a Jenkins build 7: https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-2775/

It looks like the cancelling of build 6 took longer than it took build 7 to run and complete. So my guess is the GitHub Check update from build 6 happened after build 7 had reported success.

/cc @elastic/observablt-robots I don't know at all how the apm-pipeline-library GitHub check reporting works, but it might be a nice-to-have for the reporting of the "Test" check to not report if it isn't the latest build? That might be difficult to know/coordinate, though.

**Comment by v1v** [comment]
> /cc https://github.com/orgs/elastic/teams/observablt-robots I don't know at all how the apm-pipeline-library GitHub check reporting works, but it might be a nice-to-have for the reporting of the "Test" check to not report if it isn't the latest build? That might be difficult to know/coordinate, though.

I'm a bit confused...

[build 6](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-2775/6/console) finished `19:27:12  Finished: ABORTED` since it was aborted by [build 7](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-2775/7/) that finished `16:47:34  Finished: SUCCESS`

So the time looks correct.

Therefore, https://github.com/elastic/apm-agent-nodejs/pull/2775/commits/87aff0a0fced3342e0521b6193703d4cecb6757d caused build 6 while https://github.com/elastic/apm-agent-nodejs/pull/2775/commits/405f4cce312d87e9786fa032fe155a342e214ba8 caused build 7.

The github commit status looks correct as far as I see

Builds run concurrently, but the latest one will abort all the old running ones to reduce the cost of resources while contributors keep pushing changes before the CI has finished.

As far as I see the GitHub checks reports the correct status too for https://github.com/elastic/apm-agent-nodejs/pull/2775/commits/87aff0a0fced3342e0521b6193703d4cecb6757d 

image

Otherwise, I don't know exactly what's the problem and how to reproduce it

**Review by astorm — APPROVED** [review]
👍 does the thing, approving.

File: test/instrumentation/modules/http/github-423.test.js [hunk 1]
```diff
@@ -31,7 +37,7 @@ test('https://github.com/elastic/apm-agent-nodejs/issues/423', function (t) {
     var server = http.createServer(function (req, res) {
       got(url).then(function (response) {
         t.strictEqual(response.body.length, fileSize, 'body should be expected size')
-        t.strictEqual(response.body.slice(0, 12), '\'use strict\'', 'body should be uncompressed')
+        t.strictEqual(response.body.slice(0, 12), '/*\n * Copyri', 'body should be uncompressed')
```
[the remark this answers is not in the document]
Comment (trentm) [reply test/instrumentation/modules/http/github-423.test.js:40]: Yup.

File: types/aws-lambda.d.ts [hunk 2]
```diff
@@ -7,6 +13,29 @@
 //
 // https://github.com/elastic/apm-agent-nodejs/issues/2331#issuecomment-921251030
 
+/**
```
[the remark this answers is not in the document]
Comment (trentm) [reply types/aws-lambda.d.ts:16]: Yes, and it was added to the NOTICE file as well. It was just missed before.

**Comment by v1v** [comment]
> [build 6](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-2775/6/console) finished 19:27:12 Finished: ABORTED since it was aborted by [build 7](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-2775/7/) that finished 16:47:34 Finished: SUCCESS

I somehow misread `19:27:12` as `16:27:12` then all the analysis I made was based on the wrong foundation :/

I'll track this conversation in a GitHub issue in the `apm-pipeline-library`

**Comment by trentm** [comment]
> I'll track this conversation in a GitHub issue in the `apm-pipeline-library`

Okay thanks. And as I said earlier, I've only ever seen this happen this one time, so this is very low prio IMO.

**Pull request merged**